### ISSUE TITLE
Scrolling down member list causes a crash

### ISSUE THREAD

**Issue by geekgonecrazy** [RocketChat/Rocket.Chat.Android#335]
Your Rocket.Chat.Android version: latest

I was in #dev on community server.  I opened member list (2k people) so I started scrolling.  I got maybe 3/4th down page and it crashed.  I can reproduce pretty easily.  I can hook up and provide logcat if needed

**Comment by leonardoaramaki** [comment]
@geekgonecrazy I tried this on #general at demo server where there're more than 130k users and the list won't show up. Which channel did you get this? Just #dev?

**Comment by geekgonecrazy** [comment]
@leonardoaramaki scroll speed seems to be a factor.  Flick through like you are attempting to get to a name towards the bottom

**Comment by rafaelks** [comment]
@geekgonecrazy What device and Android version are you using?

**Comment by geekgonecrazy** [comment]
Pixel XL Android 8.0

I'll grab logs.  I doubt you'll be able to reproduce in an emulator.  Generally hard to scroll like that in an emulator

**Comment by geekgonecrazy** [comment]
@leonardoaramaki / @rafaelks heres the exception I get when it crashes

```
07-01 14:52:33.550 12015 12127 E AndroidRuntime: FATAL EXCEPTION: OkHttp Dispatcher
07-01 14:52:33.550 12015 12127 E AndroidRuntime: Process: chat.rocket.android, PID: 12015
07-01 14:52:33.550 12015 12127 E AndroidRuntime: java.lang.ClassCastException: android.system.UnixSocketAddress cannot be cast to java.net.InetSocketAddress
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at libcore.io.IoBridge.getLocalInetSocketAddress(IoBridge.java:700)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at libcore.io.IoBridge.connectDetail(IoBridge.java:197)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at libcore.io.IoBridge.connect(IoBridge.java:138)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:129)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:356)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:200)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:182)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:356)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at java.net.Socket.connect(Socket.java:616)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.platform.AndroidPlatform.connectSocket(AndroidPlatform.java:63)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:223)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:149)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:192)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:121)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:100)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at chat.rocket.android.api.rest.CookieInterceptor.intercept(CookieInterceptor.java:26)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:135)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
07-01 14:52:33.550 12015 12127 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:764)
07-01 14:52:33.551 12015 12127 E UncaughtException: 
07-01 14:52:33.551 12015 12127 E UncaughtException: java.lang.ClassCastException: android.system.UnixSocketAddress cannot be cast to java.net.InetSocketAddress
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at libcore.io.IoBridge.getLocalInetSocketAddress(IoBridge.java:700)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at libcore.io.IoBridge.connectDetail(IoBridge.java:197)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at libcore.io.IoBridge.connect(IoBridge.java:138)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:129)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:356)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:200)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:182)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:356)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at java.net.Socket.connect(Socket.java:616)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.platform.AndroidPlatform.connectSocket(AndroidPlatform.java:63)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:223)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:149)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:192)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:121)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:100)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at chat.rocket.android.api.rest.CookieInterceptor.intercept(CookieInterceptor.java:26)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.RealCall$AsyncCall.execute(RealCall.java:135)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
07-01 14:52:33.551 12015 12127 E UncaughtException: 	at java.lang.Thread.run(Thread.java:764)
07-01 14:52:33.555 12015 12035 I FA      : Tag Manager is not found and thus will not be used
07-01 14:52:33.629 22866 28286 D b.a.a.a.a.a: line1: : -1
07-01 14:52:33.629 22866 28286 D b.a.a.a.a.a: SSE EVENT DATA: : -1
07-01 14:52:33.643   633   633 W netd    : type=1400 audit(0.0:322704): avc: denied { read write } for path="socket:[85097927]" dev="sockfs" ino=85097927 scontext=u:r:netd:s0 tcontext=u:r:untrusted_app_25:s0:c512,c768 tclass=unix_stream_socket permissive=0
07-01 14:52:33.720   633   633 W netd    : type=1400 audit(0.0:322705): avc: denied { read write } for path="socket:[85101355]" dev="sockfs" ino=85101355 scontext=u:r:netd:s0 tcontext=u:r:untrusted_app_25:s0:c512,c768 tclass=unix_stream_socket permissive=0
07-01 14:52:33.800   907  1721 W ActivityManager:   Force finishing activity chat.rocket.android/.activity.MainActivity
```

**Comment by Sing-Li** [comment]
Zenfone 3 Laser
Android 7.1.1

It took a while in my case, flicking the list up and down quickly, but it **did** crash  😞

**Comment by zokipirlo** [comment]
I'm not using this library, just found this crash from google while searching for a solution :)
Don't waste time on that, because it's [android bug](https://issuetracker.google.com/issues/63649622)

**Comment by luciofm** [comment]
thanks @zokipirlo, this is probably related yes...

**Comment by zokipirlo** [comment]
[Update](https://github.com/square/okhttp/issues/3438)

**Comment by rafaelks** [comment]
The issue is fixed in 1.0.20, releasing to the store today. Should be available in the next 24 hours.
